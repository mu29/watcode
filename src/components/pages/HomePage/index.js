import React from 'react'
import styled from 'styled-components'
import { Tab, TabPage, Header, GenericTemplate } from 'components'
import { ArtworkList } from 'containers'
import { palette } from 'services/style'

const StyledTab = styled(Tab)`
  margin: 1rem 0;
  border: 0.0625rem solid ${palette('gray.40')};
  background-color: ${palette('white.default')};

  @media(max-width: 768px) {
    margin: 0;
    border: none;
  }
`

const HomePage = () => (
  <GenericTemplate header={ <Header location="/" /> }>
    <StyledTab>
      <TabPage name="일간">
        <ArtworkList />
      </TabPage>
      <TabPage name="주간">
        <ArtworkList />
      </TabPage>
      <TabPage name="월간">
        <ArtworkList />
      </TabPage>
      <TabPage name="신간">
        <ArtworkList />
      </TabPage>
    </StyledTab>
  </GenericTemplate>
)

export default HomePage
