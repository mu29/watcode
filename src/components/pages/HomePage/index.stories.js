import React from 'react'
import { storiesOf } from '@storybook/react'
import HomePage from '.'

storiesOf('Pages/HomePage', module)
  .add('기본', () => (
    <HomePage />
  ))
