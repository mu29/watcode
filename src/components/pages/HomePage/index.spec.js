import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'
import HomePage from '.'

const wrap = (props = {}) => shallow(<HomePage { ...props } />)

describe('<HomePage />', () => {
  it('정상적으로 렌더링된다', () => {
    const wrapper = wrap()
    expect(wrapper).toMatchSnapshot()
  })
})
