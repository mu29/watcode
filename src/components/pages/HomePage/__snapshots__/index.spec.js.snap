// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> 정상적으로 렌더링된다 1`] = `
<GenericTemplate
  header={
    <Header
      location="/"
    />
  }
>
  <StyledTab>
    <TabPage
      name="일간"
    >
      <ArtworkList />
    </TabPage>
    <TabPage
      name="주간"
    >
      <ArtworkList />
    </TabPage>
    <TabPage
      name="월간"
    >
      <ArtworkList />
    </TabPage>
    <TabPage
      name="신간"
    >
      <ArtworkList />
    </TabPage>
  </StyledTab>
</GenericTemplate>
`;
